Automattic/vip-go-mu-plugins
# Simplifies sweep cron

## Description
This addresses a bug where we would keep scheduling a processor job on a busy cron if we would fail to get the number of already scheduled processor jobs before timeout.

## Changelog Description

### Plugin Updated: Search

We simplified logic around queued indexing of documents. The main motivation was to eliminate an edge case in which we could have scheduled more than 5 (the current limit) of indexing jobs. 

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test

1. Have multiple index versions (change of post is queued for inactive version so we will take advantage of this to add something to a queue) - `vip dev-env exec -- wp vip-search index-versions add post`
2. Add a category to a post: `vip dev-env exec -- wp post term add 1 category new-category2` (this will index the active category right away, but will queue the inactive vesion, without scheduling processor)
3. Run `dev-env exec -- wp cron event list` to verify there is **no** `vip_search_queue_processor` job
4. Run `dev-env exec -- wp cron event run vip_search_queue_sweeper` (this will create a processor job)
5.  Run `dev-env exec -- wp cron event list` to verify there **is** `vip_search_queue_processor` job
4. Run again `dev-env exec -- wp cron event run vip_search_queue_sweeper` and see it doesnt add more processor jobs
